csharp/ql/src/Security Features/CWE-134/UncontrolledFormatString.ql

```diff
@@ -0,0 +1,37 @@
+/**
+ * @name Uncontrolled format string
+ * @description Passing untrusted format strings from remote data sources can throw exceptions
+ *              and cause a denial of service.
+ * @kind path-problem
+ * @problem.severity error
+ * @precision high
+ * @id cs/uncontrolled-format-string
+ * @tags security
+ *       external/cwe/cwe-134
+ */
+
+import csharp
+import semmle.code.csharp.dataflow.flowsources.Remote
+import semmle.code.csharp.dataflow.TaintTracking
+import semmle.code.csharp.frameworks.System
+import DataFlow::PathGraph
+
+class FormatStringConfiguration extends TaintTracking::Configuration
+{
+  FormatStringConfiguration() { this = "FormatStringConfiguration" }
+  
+  override predicate isSource(DataFlow::Node source) {
+    source instanceof RemoteFlowSource
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(MethodCall call | sink.asExpr() = call.getArgumentForName("format") and
+      call.getTarget() = any(SystemStringClass s).getFormatMethod()
```

Comment:
Import `semmle.code.csharp.frameworks.Format` and do
```
exists(MethodCall call, FormatMethod fm, Parameter p |
  sink.asExpr() = call.getArgumentForParameter(p) |
  call.getTarget() = fm and
  p = fm.getParameter(fm.getFormatArgument())
)
```

Reply:
This is a really great suggestion. I found `FormatCall` which made things even simpler.